src/main/scala/beam/agentsim/agents/PersonAgent.scala

```diff
@@ -298,9 +308,12 @@ class PersonAgent(val beamServices: BeamServices, override val id: Id[PersonAgen
             }else{
               processedData.nextVehicleAssignment.beamVehicleId
             }
-            passengerSchedule.addPassenger(vehicleId,Vector(processedData.nextLeg))
+            //TODO the following needs to find all subsequent legs in currentRoute for which this agent is driver and vehicle is the same...
```

Comment:
Please note this TODO hasn't been completed but should... just scan all upcoming Legs until the BeamVehicleAssignment either says asDriver==F or the beamVehicleId does not match the vehicle she is about to BecomeDriver of.